### ISSUE TITLE
[Docs] Improve Tags Filter Interaction

### ISSUE THREAD

**Issue by crjm** [dagger/dagger#5037]
- The dropdown in the [tag filtering section](https://docs.dagger.io/278912/guides/) does not work with the keyboard only as the component uses the `onClick()` React method only to be fired.
 - Dropdown items are not selectable with `Enter` or browsable with `Tab`.
 - The input does not occupy the whole box width, so when clicking a bit to the right, nothing happens.
